code/obj/item/implant.dm

```diff
@@ -706,6 +706,39 @@ ABSTRACT_TYPE(/obj/item/implant/revenge)
 			src.owner?.elecgib()
 		. = ..()
 
+/obj/item/implant/revenge/wasp
+	name = "wasp implant"
+	big_message = " buzzes, what?"
+	small_message = "buzzes loudly, uh oh!"
+	power = 8
+	/// did we change our user's faction? this is in case a botanist gets this implant then gets it removed
+	var/changed_faction = FALSE
+
+	implanted(var/mob/M, mob/I)
+		..()
+		if (istype(M) && M.faction != FACTION_BOTANY)
```

Comment:
`faction` is a bitfield - use bitwise operators when dealing with it when appropriate